### ISSUE TITLE
Ordered Shutdown/Restart Checklist for Infrastructure Services

### ISSUE THREAD

**Issue by jermnelson** [sul-dlss/common-accessioning#710]
In anticipation of UIT network switch maintenance, DLSS systems will be shut down by midnight Saturday, 28 November and brought them back up some time after 4AM Sunday, November 29th.  This checklist is the desired ordered for Infrastructure services to be shut-down and then restarted to minimize HoneyBadger alerts.

## Julian's Plan for Ops to shut services. 
1. Development, QA, and Test systems shut-down on Friday, 13th and starting back up Monday 16h.
1. Give him a list of servers to keep up on Friday, systems not on list will be shutdown anytime business hours on Friday
   1. H2 stage and qa, dor-services-app, Fedora, Solr, common-accessioning, other robots? Ops can shutdown qa, uat, and stage servers on Friday? Devs can work locally on laptop. Inconvenient but do not want to stop Ops from their planning.
1.  Production stack with low SLA, will be shut-down after 6pm on Friday, start back up 8am Monday, SLA levels do not
    reflect dependencies between services. **ACTION** give ops a list of SLA servers
1.  Medium and High SLA will be up until  6pm Saturday, once UIT is finished, will start bringing these up
1.  Searchworks, Library website, Symphony, last to be taken down on Saturday at 10pm, up first after UIT is finished
    1.  ETD depend on SDR entire system, the thing that creates a ETD from POST in Register, Naomi guess we won't get any newly created ETD over the weekend, student who try to submit a ETD we will get an email at we need to retry on Monday. 
    1. Request Ops ETD up until 10pm on Saturday and brought up on Sunday. DB and Redis on the same VMs? Yes, we think so on etd-prod.
1. Hydrus up until 6pm on Saturday? Solr, Fedora, Suri, workflow services, dor services app, dor-indexing-app, dor-camel-routes, sul-mq, sul-robots-redis, /dor mount, data/hydrus-files storage mount
1.  This impacts all of campus, we need to have a longer time period to do this shutdown. 
1.  UIT finished at noon on Sunday, do we want to have our systems up on Sunday if we don't test over the weekend? Should we wait until Monday? When does Monday start that is early as possible. John will try to be on by 7 PST. Get a list of low-medium-high SLA per Ops? Review by the team for any problematic dependencies. Solr will get brought on Sunday to support Access.  

**DECISION** Ops can take down qa, stage, and test environments during the work-day on Friday.  Hydrus down on Saturday, ETD down on Saturday 10pm. ETD must be up on Sunday but not ETD dependencies. 

### Shutdown Steps:
- [ ] Front-end Applications
   - [ ] Google Books: enable google books via the UI: https://sul-gbooks-prod.stanford.edu/features
   - [ ] Argo 
- [ ] Worker and Robots VMs
  - [ ] Preservation Robots (preservation_robots): stop the workers 
  - [ ] Common Accessioning Robots (common-accessioning): stop the workers 
  - [ ] Pre-assembly Robots (pre-assembly): stop the workers
  - [ ] WAS Robots (was_robot_suite): stop the workers
  - [ ] GIS Robots (gis-robot-suite): stop the workers
- [ ] Service Application (DSA, DIA, techmd, etc)
- [ ] Solr
- [ ] Fedora
- [ ] Redis and Messaging-related
- [ ] Databases

### Restart Steps:
- [ ] Databases
- [ ] Redis and Messaging-related 
- [ ] Fedora
- [ ] Solr
- [ ] Service Application (DSA, DIA, techmd, etc)
- [ ] Worker and Robots VMs
  - [ ] Preservation Robots (preservation_robots): start the workers 
  - [ ] Common Accessioning Robots (common-accessioning): start the workers 
  - [ ] Pre-assembly Robots (pre-assembly): start the workers
  - [ ] WAS Robots (was_robot_suite): start the workers
  - [ ] GIS Robots (gis-robot-suite): start the workers
- [ ] Front-end Applications
   - [ ] Google Books: enable google books via the UI: https://sul-gbooks-prod.stanford.edu/features
   - [ ] Hydrus : restore the app
   - [ ] ETD
   - [ ] Argo 

db boxes, redis + messaging-related boxes, fedora, solr, service app boxes (DSA, DIA, techmd, etc.), worker + robot boxes, front-end app boxes

**Comment by peetucket** [comment]
Note that for the storage migration, I created a branch of each robot project off of master, set the worker counts to 0 in the `resque-pool.yml' file, and then deployed that specific branch (don't merge to master of course).  This was because the resque stop command wasn't working for me.  While we believe we have since fixed this issue with the rake command to stop workers, setting the workers to 0 may still be desirable to prevent the workers from starting up again when the servers are restarted (i.e. only if we are concerned about bringing them up in a specific order).

**Comment by andrewjbtw** [comment]
We should also turn off the preassembly robots. We relied on people not accessioning during the /dor migration, but it would be safer to stop them. There's a chance someone will start a job that would run through the weekend.

**Comment by andrewjbtw** [comment]
I'm still not clear how this differs from other outages. Do we need to think about turning off the workflow service too?

**Comment by peetucket** [comment]
Note that the assembly robots are included in the common-accessioning codebase.  Technically the pre-assembly workers aren't associated with robots, but the same principle should apply (you can set the worker count to 0 in the `resque-pool.yml` file or stop resque with the rake command.

**Comment by mjgiarlo** [comment]
Other sidekiq queues that may need quiescing (graceful stop):

- [ ] argo
- [ ] technical-metadata-service
- [ ] google-books
- [ ] dor-services-app
- [ ] sdr-api
- [ ] hydra_etd
- [ ] was-registrar-app
- [ ] happy-heron

**Comment by ndushay** [comment]
See https://github.com/sul-dlss/DevOpsDocs/pull/479

**Comment by jermnelson** [comment]
Closing, superseded by https://github.com/sul-dlss/DevOpsDocs/pull/479
